### PR TITLE
Fix split editor view with the new editor

### PR DESCRIPTION
**Pull request by sevoku**
Use the same technique we use for Pad resizing in #7762 to fix the resizing in the document split view.

> Fixes VSTS [#910879](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/910879)

### CODE REVIEW

**Comment by sevoku** [comment]
This approach has several issues:

Rendering:
grafik
It looks like the splitter is not transparent above Gtk widgets. In the screenshot the editor area is rendered above (see how the breakpoint overlaps the splitter area), but the header are (tabs) is behind the splitter widget.

Events:
Presumably because the editors are added after the splitter, they are above it, so the splitter is not receiving all events. It's similar to https://github.com/mono/monodevelop/pull/7762#issuecomment-499548257, but I have no idea how to solve this here.

My feeling is that both issues are related. We need to find a way to add the splitter on the very top, above Gtk and native, but make it transparent at the same time. @abock do you have an idea?

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Components/HPanedThin.cs [hunk 1]
```diff
@@ -88,19 +97,154 @@ protected override bool OnExposeEvent (Gdk.EventExpose evnt)
 		}
 	}
 
-	class CustomPanedHandle: Gtk.EventBox
+#if MAC
+
+	class CustomMacPanedHandle : CustomPanedHandle
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Components/HPanedThin.cs]:
```suggestion
	sealed class CustomMacPanedHandle : CustomPanedHandle
```

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Components/HPanedThin.cs [hunk 2]
```diff
@@ -88,19 +97,154 @@ protected override bool OnExposeEvent (Gdk.EventExpose evnt)
...
+				if (horizontal)
+					SizeAllocate (new Gdk.Rectangle (parent.Child1.Allocation.X + parent.Child1.Allocation.Width - centerSize, args.Allocation.Y, GrabAreaSize, args.Allocation.Height));
+				else
+					SizeAllocate (new Gdk.Rectangle (args.Allocation.X, parent.Child1.Allocation.Y + parent.Child1.Allocation.Height - centerSize, args.Allocation.Width, GrabAreaSize));
+			} else
+				Hide ();
+		}
+	}
+
+	class MacPanedHandleView: DragEventTrapView
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Components/HPanedThin.cs]:
```suggestion
	sealed class MacPanedHandleView: DragEventTrapView
```

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Components/HPanedThin.cs [hunk 3]
```diff
@@ -88,19 +97,154 @@ protected override bool OnExposeEvent (Gdk.EventExpose evnt)
...
+			VisibleWindow = false;
+			this.parent = parent;
+			HPanedThin.InitStyle (parent, 1);
+			horizontal = parent is HPanedThin;
+
+			handle = new MacPanedHandleView (parent);
+			host = new Gtk.GtkNSViewHost (handle, NSWindowOrderingMode.Above);
+
+			GrabAreaSize = HandleGrabWidth;
+			parent.SizeRequested += delegate {
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Components/HPanedThin.cs]: ref cycle.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Components/HPanedThin.cs [hunk 4]
```diff
@@ -88,19 +97,154 @@ protected override bool OnExposeEvent (Gdk.EventExpose evnt)
...
+			base.MouseDown (theEvent);
+		}
+
+		public override void MouseDragged (NSEvent theEvent)
+		{
+			base.MouseDragged (theEvent);
+
+			var point = NSEvent.CurrentMouseLocation;
+
+			if (horizontal) {
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Components/HPanedThin.cs]:
```csharp
int relativeTo = horizontal ? (int)point.X : (int)point.Y;
int newpos = initialPanedPos + relativeTo - initialPos;
owner.Position = newpos >= 10 ? newpos : 10;
```

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Components/HPanedThin.cs [hunk 5]
```diff
@@ -88,19 +97,154 @@ protected override bool OnExposeEvent (Gdk.EventExpose evnt)
...
+	abstract class CustomPanedHandle: Gtk.EventBox
+	{
+		internal const int HandleGrabWidth = 4;
+
+		abstract public int GrabAreaSize { get; set; }
+
+		virtual public Gtk.Widget HandleWidget { get; set; }
+	}
+
+	class CustomGtkPanedHandle: CustomPanedHandle
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Components/HPanedThin.cs]:
```suggestion
	sealed class CustomGtkPanedHandle: CustomPanedHandle
```

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Components/HPanedThin.cs [hunk 6]
```diff
@@ -88,19 +97,154 @@ protected override bool OnExposeEvent (Gdk.EventExpose evnt)
...
+		}
+	}
+
+#endif
+
+	abstract class CustomPanedHandle: Gtk.EventBox
+	{
+		internal const int HandleGrabWidth = 4;
+
+		abstract public int GrabAreaSize { get; set; }
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Components/HPanedThin.cs]: Style nit:
```suggestion
	public abstract int GrabAreaSize { get; set; }
```

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Components/HPanedThin.cs [hunk 7]
```diff
@@ -88,19 +97,154 @@ protected override bool OnExposeEvent (Gdk.EventExpose evnt)
...
+
+		virtual public Gtk.Widget HandleWidget { get; set; }
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Components/HPanedThin.cs]: Style nit:
```suggestion
		public virtual Gtk.Widget HandleWidget { get; set; }
```

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Components/HPanedThin.cs [hunk 8]
```diff
@@ -88,19 +97,154 @@ protected override bool OnExposeEvent (Gdk.EventExpose evnt)
...
+		{
+			VisibleWindow = false;
+			this.parent = parent;
+			HPanedThin.InitStyle (parent, 1);
+			horizontal = parent is HPanedThin;
+
+			handle = new MacPanedHandleView (parent);
+			host = new Gtk.GtkNSViewHost (handle, NSWindowOrderingMode.Above);
+
+			GrabAreaSize = HandleGrabWidth;
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Components/HPanedThin.cs]: HandleGrabWidth is only used here.

**Comment by sevoku** [comment]
@monojenkins backport release-8.2

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Components/HPanedThin.cs [hunk 9]
```diff
@@ -88,46 +97,221 @@ protected override bool OnExposeEvent (Gdk.EventExpose evnt)
...
+		}
+
+		void HandleSizeAllocated (object o, Gtk.SizeAllocatedArgs args)
+		{
+			OnParentSizeAllocated (args);
+		}
+
+		protected override void OnDestroyed ()
+		{
+            if (ParentPaned != null) {
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Components/HPanedThin.cs]:
```suggestion
			if (ParentPaned != null) {
```

**Pull request merged**